vividus-allure-adaptor/src/main/java/org/vividus/bdd/report/allure/AllureStoryReporter.java

```diff
@@ -441,6 +442,23 @@ public void onAttachmentPublish(AttachmentPublishEvent event)
         lifecycle.addAttachment(attachment.getTitle(), attachment.getContentType(), null, attachment.getContent());
     }
 
+    @Subscribe
+    public void onLinkPublish(LinkPublishEvent event)
+    {
+        lifecycle.updateTestCase(getRootStepId(), result ->
+        {
+            boolean exists = result.getLinks().stream()
+                                              .anyMatch(l -> l.getUrl().equals(event.getUrl())
+                                                      && l.getName().equals(event.getName()));
+
+            if (!exists)
+            {
+                Link link = ResultsUtils.createLink(null, event.getName(), event.getUrl(), "tms");
```

Comment:
why `tms`?

Reply:
there is no much variety, either issue or tms, we can consider introducing new icon for links in the future

Reply:
you can use any link type, the difference is that custom link types won't have icons

Reply:
yes, but this will look like plain txt, that's why I excluded this from the choice

Reply:
TMS - Test management system, not every link points to TMS, so the event should have ability to provide link type, it won't have any icon, but that's better than misleading icon

Reply:
I believe that for end users icons just make the report more readable by separating different links, also icon of database it too abstract to refer to TMS, it may be even more misleading

Reply:
https://github.com/allure-framework/allure2/blob/master/allure-generator/src/main/javascript/plugins/testresult-links/LinksView.hbs